### ISSUE TITLE
Overuse of property search hooks???

### ISSUE THREAD

**Issue by markheydon** [propertyhive/WP-Property-Hive#51]
Hi,

I noticed this while looking at the possibility of adding a new shortcode.  It seems that the propertyhive_before_search_results_loop_item and propertyhive_after_search_results_loop_item hooks are present in the following files.

- templates/content-property-featured.php
- templates/content-property-recent.php
- templates/content-property.php

Shouldn't the first two be featured property and recent property hooks instead?

I'm happy to submit a PR to change them if so.

Mark

**Comment by propertyhive** [comment]
@markheydon You're absolutely right. It's something which I've noticed before and wanted to put right at some point. 

As with most things when it comes to changing pre-existing functionality, my concern relates to how to do this in such a way that existing sites don't get effected.

In relation to this, we should also have a 'content-property-similar.php' which is included when the [similar_properties] shortcode is used. This is another which I've noticed, but the consequences of breaking existing sites has put it on the backburner.

Have you any experience with this? Part of me thinks just leave it, it's not worth the hassle, and the other part of me thinks just put it out there with an announcement/upgrade warning etc so people know to check things first before updating.

I'm happy to make the change this end, just needs some forward planning on our part :)

Thanks for the feed back though!
Steve

**Comment by markheydon** [comment]
Hey Steve (@propertyhive),

Good to hear from you again -- we've had some emails in the past -- I did think of the 'breaking change' issue, but honestly didn't have time to fully evaluate it.  I was in the middle of trying to get my head around the various filter when I noticed it.

I'm in the middle of adding a shortcode for a quick/simple featured property carousel re-using the flixslider stuff already in there for the property detail page (for which I'll open a separate issue for discussion later today).  While I'm adding the filters for that this afternoon, I will do some quick analysis of the full impact of making the change and report back on both.

I wasn't completely sure this project was still being maintained, to be honest as I found some room related file(s) -- e.g. `templates/single-property/rooms.php` -- in the official version of the plugin that doesn't appear to be in the GitHub repo?  It was late last night though so I may have been looking at the wrong thing!

I'm currently on the hunt for a couple of open source projects to get involved with by the way so more than happy to put a bit of my time into this one if you are looking for contributors at this time?

Mark

**Comment by markheydon** [comment]
Hi Steve (@propertyhive),

I did a full search for all the actions and found the following four (that we knew about already) that didn't have corresponding do_action() calls.

- propertyhive_before_featured_loop_item_title
- propertyhive_after_featured_loop_item_title
- propertyhive_before_recent_loop_item_title
- propertyhive_after_recent_loop_item_title

If you take the view that if anybody had created actions against any of those, they wouldn't have worked anyway and that it's sensible to have the featured property and recent property to continue to work out-of-the-box as it currently does, what about this as a possible workaround... #54.

Basically, I'm suggesting just remove the references from `ph-template-hooks.php` and have them fall back to a default if they don't exist, which is likely they don't in the wild as it wasn't working anyway.

Mark
